package.json

```diff
@@ -25,10 +25,21 @@
     "onLanguage:reason"
   ],
   "icon": "assets/logo.png",
+  "main": "./dist/extension.js",
   "contributes": {
     "configuration": {
       "title": "OCaml configuration",
       "properties": {
+        "ocaml.toolChainRoot": {
+          "type": "string",
+          "default": null,
+          "description": "Determines where to find the toolchain for a given project"
+        },
+        "ocaml.packageManager": {
```

Comment:
Given the following snippet from `Toolchain.ml`
```ocaml
let ofName ~root = function
    | "opam" -> Some (Opam root)
    | "esy" -> Some (Esy root)
    | "global" -> Some Global
    | _ -> None
```
I would add
```json
"enum": [ "esy", "opam", "global" ]
```
to the `ocaml.packageManager` configuration. Somehow this is not documented on the official vscode website, but `enum` allows to have a dropdown selection in the vscode configuration menu

Reply:
I'm not sure why we need both this option and the toolchain root actually. The two options seem redundant to me. What about something like a single option interpreted as follows:

* null - the global toolchain is used
* a file that ends with json - esy is used
* a directory or a raw name will be interpreted as an opam switch

Reply:
@rgrinberg We need to provide a way to override the choice of package manager/toolchain manager. Dropdown/explicit field in the settings is a great way to do this, IMO, and not redundant.

The reason `toolchainRoot` exists there is not for users to override per se - vscode's settings persistence API requires us to expose the field for users to edit.